validations/src/validations.rs

```diff
@@ -45,45 +49,51 @@ use witnet_rad::{
 };
+use witnet_stack::{execute_complete_script, Item, MyValue};
 
-/// Returns the fee of a value transfer transaction.
+/// Returns the fee of a transaction.
 ///
 /// The fee is the difference between the outputs and the inputs
 /// of the transaction. The pool parameter is used to find the
 /// outputs pointed by the inputs and that contain the actual
 /// their value.
-pub fn vt_transaction_fee(
-    vt_tx: &VTTransaction,
+pub fn transaction_fee(
+    tx: &Transaction,
     utxo_diff: &UtxoDiff<'_>,
     epoch: Epoch,
     epoch_constants: EpochConstants,
 ) -> Result<u64, failure::Error> {
-    let in_value = transaction_inputs_sum(&vt_tx.body.inputs, utxo_diff, epoch, epoch_constants)?;
-    let out_value = transaction_outputs_sum(&vt_tx.body.outputs)?;
+    let (in_value, out_value) = match tx {
+        Transaction::ValueTransfer(vt_tx) => {
+            let in_value =
+                transaction_inputs_sum(&vt_tx.body.inputs, utxo_diff, epoch, epoch_constants)?;
+            let out_value = transaction_outputs_sum(&vt_tx.body.outputs)?;
 
-    if out_value > in_value {
-        Err(TransactionError::NegativeFee.into())
-    } else {
-        Ok(in_value - out_value)
-    }
-}
+            (in_value, out_value)
+        }
+        Transaction::Script(sh_tx) => {
+            let inputs: Vec<Input> = sh_tx
+                .body
+                .inputs
+                .iter()
+                .map(|sh_input| sh_input.to_input())
+                .collect();
+            let in_value = transaction_inputs_sum(&inputs, utxo_diff, epoch, epoch_constants)?;
+            let out_value = transaction_outputs_sum_script(&sh_tx.body.outputs)?;
 
-/// Returns the fee of a data request transaction.
-///
-/// The fee is the difference between the outputs (with the data request value)
-/// and the inputs of the transaction. The pool parameter is used to find the
-/// outputs pointed by the inputs and that contain the actual
-/// their value.
-pub fn dr_transaction_fee(
-    dr_tx: &DRTransaction,
-    utxo_diff: &UtxoDiff<'_>,
-    epoch: Epoch,
-    epoch_constants: EpochConstants,
-) -> Result<u64, failure::Error> {
-    let in_value = transaction_inputs_sum(&dr_tx.body.inputs, utxo_diff, epoch, epoch_constants)?;
-    let out_value = transaction_outputs_sum(&dr_tx.body.outputs)?
-        .checked_add(dr_tx.body.dr_output.checked_total_value()?)
-        .ok_or(TransactionError::OutputValueOverflow)?;
+            (in_value, out_value)
+        }
+        Transaction::DataRequest(dr_tx) => {
+            let in_value =
+                transaction_inputs_sum(&dr_tx.body.inputs, utxo_diff, epoch, epoch_constants)?;
+            let out_value = transaction_outputs_sum(&dr_tx.body.outputs)?
+                .checked_add(dr_tx.body.dr_output.checked_total_value()?)
+                .ok_or(TransactionError::OutputValueOverflow)?;
+
+            (in_value, out_value)
+        }
+        _ => unreachable!("Only VT, DR and script transactions allowed"),
```

Comment:
```suggestion
        _ => panic!("Only VT, DR and script transactions allowed"),
```

Because there is nothing preventing users from calling `transaction_fee` with other kinds of transactions.